src/amo/reducers/home.js

```diff
@@ -8,49 +8,113 @@ import {
 import { SET_CLIENT_APP } from 'core/constants';
 import { createInternalAddon } from 'core/reducers/addons';
 import { isTheme } from 'core/utils';
-import type { AddonType, ExternalAddonType } from 'core/types/addons';
 import type { SetClientAppAction } from 'core/actions';
+import type {
+  AddonType,
+  ExternalAddonType,
+  PartialExternalAddonType,
+} from 'core/types/addons';
+
+export const FETCH_HOME_DATA: 'FETCH_HOME_DATA' = 'FETCH_HOME_DATA';
+export const LOAD_HOME_DATA: 'LOAD_HOME_DATA' = 'LOAD_HOME_DATA';
+
+export type PrimaryHeroShelfExternalType = {|
+  id: string,
+  guid: string,
+  homepage: string,
+  name: string,
+  type: string,
+|};
+
+export type ExternalPrimaryHeroShelfType = {|
+  gradient: {|
+    start: string,
+    end: string,
+  |},
+  featured_image: string,
+  description: string | null,
+  addon: PartialExternalAddonType | void,
+  external: PrimaryHeroShelfExternalType | void,
+|};
+
+export type PrimaryHeroShelfType = {|
+  gradient: {|
+    start: string,
+    end: string,
+  |},
+  featuredImage: string,
+  description: string | null,
+  addon: AddonType | void,
+  external: PrimaryHeroShelfExternalType | void,
+|};
+
+export type HeroCTAType = {|
+  url: string,
+  text: string,
+|};
+
+export type SecondaryHeroModuleType = {|
+  icon: string,
+  description: string,
+  cta: HeroCTAType,
```

Comment:
`cta` might be `null` according to the API doc: https://addons-server.readthedocs.io/en/latest/topics/api/hero.html#secondary-hero-shelves